components/backup/src/endpoint.rs

```diff
@@ -438,6 +503,13 @@ impl<E: Engine, R: RegionInfoProvider> Runnable<Task> for Endpoint<E, R> {
     }
 }
 
+impl<E: Engine, R: RegionInfoProvider> RunnableWithTimer<Task, ()> for Endpoint<E, R> {
+    fn on_timeout(&mut self, timer: &mut Timer<()>, _: ()) {
+        self.pool.borrow_mut().check_active();
```

Comment:
Could you test whether the pool can be shutdown?

Reply:
Sure.